src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -267,16 +269,33 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
             {                
                 return await asyncConnectorTexlFunction.InvokeAsync(args, _services, _cancellationToken).ConfigureAwait(false);
             }
-            else if (func is UserDefinedTexlFunction udtf)
-            {
-                // $$$ Should add _runtimeConfig
-                result = await udtf.InvokeAsync(args, _cancellationToken, context.StackDepthCounter.Increment()).ConfigureAwait(false);
-            }
             else if (func is CustomTexlFunction customTexlFunc)
             {
                 // If custom function throws an exception, don't catch it - let it propagate up to the host.
                 result = await customTexlFunc.InvokeAsync(FunctionServices, args, _cancellationToken).ConfigureAwait(false);
             }
+            else if (func is UserDefinedFunction userDefinedFunc)
+            {
+                var recordValue = userDefinedFunc.GetRuntimeRecordValue(args, _cancellationToken);
+
+                var symbolsTableRecord = ReadOnlySymbolTable.NewFromRecord(recordValue.Type);
+                var symbolValuesRecord = SymbolValues.NewFromRecord(symbolsTableRecord, recordValue);
+
+                var symbolsTable = ReadOnlySymbolTable.Compose(_symbolValues.SymbolTable, symbolsTableRecord);
+                var symbolValues = ReadOnlySymbolValues.Compose(_symbolValues, symbolValuesRecord);
+
+                userDefinedFunc.SetFormulaValues(args);
+                userDefinedFunc.BindBody(symbolsTable, new Glue2DocumentBinderGlue(), bindingConfig: BindingConfig.Default);
+
+                (var irnode, var ruleScopeSymbol) = userDefinedFunc.GetIRTranslator();
+
+                var expr = new ParsedExpression(irnode, ruleScopeSymbol, context.StackDepthCounter)
+                {
+                    _allSymbols = symbolsTable
+                };
+                
+                result = expr.Eval(symbolValues);

```

Comment:
>expr.Eval(symbolValues); [](http://example.com/codeflow?start=25&length=24)

I think we should pass here runtime config with local?
Also, I would use Async api here.